chainstore/chainstore.go

```diff
@@ -110,7 +114,8 @@ func (s *Store) load(tsk types.TipSetKey, v interface{}) error {
 		return fmt.Errorf("error getting key %s: %s", key, err)
 	}
 	if err := json.Unmarshal(buf, v); err != nil {
-		return err
+		log.Warnf("corrupted checkpoint: %s", buf)
```

Comment:
So somehow here `buf` is a plain string (mentioned in PR description) instead of a `chainIndex` json.